[UI] Workaround layering model

andersonjeccel opened the pull request:
<!-- ## Which branch should I use for my PR?

Assuming that:

a = current major release
b = current minor release
c = future major release

* a.x for any features and enhancements (e.g. 5.x)
* a.b for any bug fixes (e.g. 4.4, 5.1)
* c.x for any features, enhancements or bug fixes with backward compatibility breaking changes (e.g. 5.x) -->

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🟢 <!-- Use emojis to indicate positive (green) or negative (red) for each item in the table. -->
| New feature/enhancement? (use the a.x branch)      | 🔴
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🔴 <!-- All PRs must maintain or improve code coverage -->
| Related user documentation PR URL      | mautic/user-documentation#... <!-- required for new features -->
| Related developer documentation PR URL | mautic/developer-documentation-new#... <!-- required for developer-facing changes -->
| Issue(s) addressed                     | Fixes #... <!-- prefix each issue number with "Fixes #", no need to create an issue if none exists, explain below instead -->

<!--
Additionally (see https://contribute.mautic.org/contributing-to-mautic/developer/code/pull-requests#work-on-your-pull-request):
 - Always add tests and ensure they pass.
 - Bug fixes must be submitted against the lowest maintained branch where they apply
   (lowest branches are regularly merged to upper ones so they get the fixes too.)
 - Features and deprecations must be submitted against the "4.x" branch.
-->

## Description

This PR changes the CSS structure related to the layering model and wipes automatic styles that weren't working properly. Also includes a temporary workaround: tab content bg was reset to transparent as fix for the issue mentioned [in Slack](https://mautic.slack.com/archives/C024XBCQS3Z/p1732115616475359) by @laurielim.

<!--
Please write a short README for your feature/bugfix. This will help people understand your PR and what it aims to do. If you are fixing a bug and if there is no linked issue already, please provide steps to reproduce the issue here.
-->
<!-- Remove HTML comment markup below to use the table for screenshots when relevant. -->
| Before                                 | After
| -------------------------------------- | ---
| ![image](https://github.com/user-attachments/assets/d580d637-9fb9-4f63-882a-ba67c2fb3367)                                     | ![image](https://github.com/user-attachments/assets/e58df225-75a9-47b8-85c8-24c716ee487f)


---
### 📋 Steps to test this PR:

<!--
This part is crucial. Take the time to write very clear, annotated and step by step test instructions, because testers may not be developers.
-->
1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Open Forms
3. Create new
4. See that fields are visible on the screen without background

<!--
If you have any deprecations and backwards compatibility breaks, list them here along with the new alternative.
-->

andersonjeccel committed:
Resetting layering variables name

andersonjeccel committed:
Adding layering model for 2 steps

andersonjeccel committed:
Replacing old layering name

andersonjeccel committed:
bdr-r

andersonjeccel committed:
fixing nav tabs bdr radius

andersonjeccel committed:
fixing input btn last child border radius

andersonjeccel committed:
Fixing excessive bdr radius for code items

andersonjeccel committed:
Fixing copy icon alignment and border radius

andersonjeccel committed:
Fixing dropdown items bdr r

andersonjeccel committed:
Fixing list group items bdr r

andersonjeccel committed:
Removing broken automatic layering

andersonjeccel committed:
giving up on modals with light gray bg

andersonjeccel committed:
temporary workaround for layering in tab content

laurielim reviewed (approved):
It works as intended and the fields are now visible.

I found one odd things although I don't think it's a blocker. When using autocomplete to fill a field, somehow the background color of the input changes (see "Form attributes" in the image below) but this does not happen if you type something in.
![image](https://github.com/user-attachments/assets/0dde75ba-c922-47ec-81d0-e9c01917e0f2)

andersonjeccel commented:
@laurielim we really need to update bootstrap :/

shinde-rahul commented on app/bundles/CoreBundle/Assets/css/app.css:
```diff
@@ -7972,19 +7950,19 @@ ul.line-legend li {
 }
 lesshat-selector {
-  -lh-property: 0; }
+  -lh-property: 0; } 
```
```suggestion
  -lh-property: 0; }
```
The appending space is the only change. So, suggesting a revert.

andersonjeccel replied:
I'm confused about those changes, as the command `grunt compile-less` is the responsible for generating the final CSS file (so it's never manually touched)

shinde-rahul commented on app/bundles/CoreBundle/Assets/css/app.css:
```diff
@@ -7972,19 +7950,19 @@ ul.line-legend li {
 @-moz-keyframes after-anim{ 0% { width: 0%; left: 0%; } 10% { width: 30%; left: 100%; } 19.99% { width: 30%; left: 100%; } 20% { width: 0%; left: 0%; } 30% { width: 80%; left: 100%; } 30.01% { width: 0%; left: 0%; } 40% { width: 5%; left: 30%; } 50% { width: 50%; left: 100%; } 50.01% { width: 0%; left: 0%; } 60% { width: 60%; left: 20%; } 70% { width: 5%; left: 100%; } 70.01% { width: 0%; left: 0 ...
 @-o-keyframes after-anim{ 0% { width: 0%; left: 0%; } 10% { width: 30%; left: 100%; } 19.99% { width: 30%; left: 100%; } 20% { width: 0%; left: 0%; } 30% { width: 80%; left: 100%; } 30.01% { width: 0%; left: 0%; } 40% { width: 5%; left: 30%; } 50% { width: 50%; left: 100%; } 50.01% { width: 0%; left: 0%; } 60% { width: 60%; left: 20%; } 70% { width: 5%; left: 100%; } 70.01% { width: 0%; left: 0%; ...
-@keyframes after-anim{ 0% { width: 0%; left: 0%; } 10% { width: 30%; left: 100%; } 19.99% { width: 30%; left: 100%; } 20% { width: 0%; left: 0%; } 30% { width: 80%; left: 100%; } 30.01% { width: 0%; left: 0%; } 40% { width: 5%; left: 30%; } 50% { width: 50%; left: 100%; } 50.01% { width: 0%; left: 0%; } 60% { width: 60%; left: 20%; } 70% { width: 5%; left: 100%; } 70.01% { width: 0%; left: 0%; }  ...
+@keyframes after-anim{ 0% { width: 0%; left: 0%; } 10% { width: 30%; left: 100%; } 19.99% { width: 30%; left: 100%; } 20% { width: 0%; left: 0%; } 30% { width: 80%; left: 100%; } 30.01% { width: 0%; left: 0%; } 40% { width: 5%; left: 30%; } 50% { width: 50%; left: 100%; } 50.01% { width: 0%; left: 0%; } 60% { width: 60%; left: 20%; } 70% { width: 5%; left: 100%; } 70.01% { width: 0%; left: 0%; }  ...
```
```suggestion
@keyframes after-anim{ 0% { width: 0%; left: 0%; } 10% { width: 30%; left: 100%; } 19.99% { width: 30%; left: 100%; } 20% { width: 0%; left: 0%; } 30% { width: 80%; left: 100%; } 30.01% { width: 0%; left: 0%; } 40% { width: 5%; left: 30%; } 50% { width: 50%; left: 100%; } 50.01% { width: 0%; left: 0%; } 60% { width: 60%; left: 20%; } 70% { width: 5%; left: 100%; } 70.01% { width: 0%; left: 0%; } 80% { width: 50%; left: 30%; } 90% { width: 10%; left: 80%; } 100% { width: 20%; left: 100%; }}
```
Same here.

shinde-rahul commented on app/bundles/CoreBundle/Assets/css/app.css:
```diff
@@ -7972,19 +7950,19 @@ ul.line-legend li {
 }
 lesshat-selector {
-  -lh-property: 0; }
+  -lh-property: 0; } 
```
```suggestion
  -lh-property: 0; }
```

shinde-rahul commented on app/bundles/CoreBundle/Assets/css/app.css:
```diff
@@ -7972,19 +7950,19 @@ ul.line-legend li {
 @-moz-keyframes before-anim{ 0% { width: 0%; left: 0%; } 0% { width: 60%; } 19.99% { width: 40%; left: 100%; } 20% { width: 0%; left: 0%; } 25% { width: 10%; left: 10%; } 30% { width: 40%; left: 30%; } 40% { width: 60%; left: 100%; } 40.01% { width: 0%; left: 0%; } 50% { width: 10%; left: 40%; } 60% { width: 30%; left: 100%; } 60.01% { width: 0%; left: 0%; } 70% { width: 10%; left: 40%; } 80% { w ...
 @-o-keyframes before-anim{ 0% { width: 0%; left: 0%; } 0% { width: 60%; } 19.99% { width: 40%; left: 100%; } 20% { width: 0%; left: 0%; } 25% { width: 10%; left: 10%; } 30% { width: 40%; left: 30%; } 40% { width: 60%; left: 100%; } 40.01% { width: 0%; left: 0%; } 50% { width: 10%; left: 40%; } 60% { width: 30%; left: 100%; } 60.01% { width: 0%; left: 0%; } 70% { width: 10%; left: 40%; } 80% { wid ...
-@keyframes before-anim{ 0% { width: 0%; left: 0%; } 0% { width: 60%; } 19.99% { width: 40%; left: 100%; } 20% { width: 0%; left: 0%; } 25% { width: 10%; left: 10%; } 30% { width: 40%; left: 30%; } 40% { width: 60%; left: 100%; } 40.01% { width: 0%; left: 0%; } 50% { width: 10%; left: 40%; } 60% { width: 30%; left: 100%; } 60.01% { width: 0%; left: 0%; } 70% { width: 10%; left: 40%; } 80% { width: ...
+@keyframes before-anim{ 0% { width: 0%; left: 0%; } 0% { width: 60%; } 19.99% { width: 40%; left: 100%; } 20% { width: 0%; left: 0%; } 25% { width: 10%; left: 10%; } 30% { width: 40%; left: 30%; } 40% { width: 60%; left: 100%; } 40.01% { width: 0%; left: 0%; } 50% { width: 10%; left: 40%; } 60% { width: 30%; left: 100%; } 60.01% { width: 0%; left: 0%; } 70% { width: 10%; left: 40%; } 80% { width: ...
```
```suggestion
@keyframes before-anim{ 0% { width: 0%; left: 0%; } 0% { width: 60%; } 19.99% { width: 40%; left: 100%; } 20% { width: 0%; left: 0%; } 25% { width: 10%; left: 10%; } 30% { width: 40%; left: 30%; } 40% { width: 60%; left: 100%; } 40.01% { width: 0%; left: 0%; } 50% { width: 10%; left: 40%; } 60% { width: 30%; left: 100%; } 60.01% { width: 0%; left: 0%; } 70% { width: 10%; left: 40%; } 80% { width: 5%; left: 100%; } 80.01% { width: 0%; left: 0%; } 90% { width: 70%; left: 10%; } 100% { width: 10%; left: 100%; }}
```

shinde-rahul reviewed (changes requested):
Some CS-related small change requests. Otherwise, the changes look good.

andersonjeccel committed:
Merge branch '5.x' into workaround-layering-model

escopecz commented on app/bundles/CoreBundle/Assets/css/app/less/components/navs.less:
```diff
@@ -171,7 +171,7 @@
 .nav-tabs-wrapper:has(.nav-tabs-contained) + form > .tab-content,
 .nav-tabs-contained + .tab-content,
 .nav-tabs-contained + form > .tab-content {
-  background-color: var(--layer);
+  // background-color: var(--layer);
 }
```
Can this be removed when it's not needed?

andersonjeccel replied:
I'll need the selectors for the PR that solves permanently :)

andersonjeccel replied:
this is a quick-fix to allow releasing the next version, because the permanent solution involves recreating all pages that have tabs to add wrappers (I'll do it using a template then + follow the right Bootstrap HTML structure)

escopecz replied:
The comment has the same effect as deleting the 3 rows, right? I don't get why it's needed.

andersonjeccel replied:
I'll bring the snippet to a Google Doc and remove from the codebase

andersonjeccel replied:
done btw

andersonjeccel committed:
removing selectors

.nav-tabs-wrapper:has(.nav-tabs-contained) + .tab-content,
.nav-tabs-wrapper:has(.nav-tabs-contained) + form > .tab-content,
.nav-tabs-contained + .tab-content,
.nav-tabs-contained + form > .tab-content {
  // background-color: var(--layer);
}

kuzmany reviewed (approved):
Good to me 👍

matbcvo reviewed (approved):
Changes reviewed & PR tested - looks good to me.
